modules/ml/src/main/java/org/apache/ignite/ml/math/distributed/CacheUtils.java

```diff
@@ -523,16 +601,54 @@ else if (key instanceof RowColMatrixKey)
 
             return a;
         });
-        totalRes.add(defRes);
-        return totalRes.stream().reduce(zeroVal, accumulator);
+        return totalRes.stream().reduce(defRes, accumulator);
+    }
+
+    public static <K, V, A, W> A reduce(String cacheName, Ignite ignite,
```

Comment:
The main problem with this class is amount of test for all these sparse methods.
Please provide tests for new methods (which was added by you)